test/pokemon-router-test.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+require('dotenv').load();
+const superagent = require('superagent');
+const expect = require('chai').expect;
+const Pokemon = require('../model/pokemon.js');
+const baseURL = `http://localhost:${process.env.PORT || 3000}`;
+require('../server.js');
+
+describe('testing pokemon router', function() {
+  // let tempPokemon ={
+  //   name: 'Ivysaur',
+  //   type: 'grass',
+  //   moves:'cut',
+  // };
+  describe('testing POST /api/pokemon', function() {
+    after((done) => {
+      Pokemon.deleteById(this.tempPokemonId)
+      .then(() => done())
+      .catch(done);
+    });
+    it('valid input should create a pokemon', (done) => {
+      superagent.post(`${baseURL}/api/pokemon`)
+      .send({
+        name: 'Ivysaur',
+        type: 'grass',
+        moves:'cut',
+      })
+      .then(res => {
+        expect(res.status).to.equal(200);
+        expect(res.body.name).to.equal('Ivysaur');
+        expect(res.body.type).to.equal('grass');
+        expect(Boolean(res.body.id)).to.equal(true);
+        this.tempPokemonId = res.body.id;
+        done();
+      })
+      .catch(done);
+    });
+    it('invalid input should return 400 status code', (done) => {
+      superagent.post(`${baseURL}/api/pokemon`)
+      .send({name: 'Charizard'})
+      .then(done)
+      .catch(err => {
+        expect(err.status).to.equal(400);
+        done();
+      })
+      .catch(done);
+    });
+  });
+  describe('testing GET /api/pokemon/:id', function() {
+    before((done) => {
+      superagent.post(`${baseURL}/api/pokemon`)
```

Comment:
Not a point loser, but Duncan advised during a lecture that ``before`` blocks should just utilize your storage module methods rather than an actual POST request through the server. I.e. your ``storage.createItem()`` method would go here instead of making a real POST to the server.